src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/ComparisonHelper.java

```diff
@@ -89,15 +92,25 @@ public static boolean equals(AddressDto addressDto,
             return nullValuesCheck;
         }
 
-        return StringUtils.equalsIgnoreCase(normalise(addressDto.getPropertyNameNumber()), normalise(address.getPremises()))
-                && StringUtils.equalsIgnoreCase(normalise(addressDto.getLine1()), normalise(address.getAddressLine1()))
-                && StringUtils.equalsIgnoreCase(normalise(addressDto.getLine2()), normalise(address.getAddressLine2()))
-                && StringUtils.equalsIgnoreCase(normalise(addressDto.getTown()), normalise(address.getLocality()))
-                && StringUtils.equalsIgnoreCase(normalise(addressDto.getCounty()), normalise(address.getRegion()))
-                && StringUtils.equalsIgnoreCase(normalise(addressDto.getCountry()), normalise(address.getCountry()))
-                && StringUtils.equalsIgnoreCase(normalise(addressDto.getPoBox()), normalise(address.getPoBox()))
-                && StringUtils.equalsIgnoreCase(normalise(addressDto.getCareOf()), normalise(address.getCareOf()))
-                && StringUtils.equalsIgnoreCase(normalise(addressDto.getPostcode()), normalise(address.getPostalCode()));
+        String addressDtoLine1 = combineAddressLines(addressDto.getPropertyNameNumber(),
+                addressDto.getLine1());
+        String addressLine1 = combineAddressLines(address.getPremises(), address.getAddressLine1());
+
+        return StringUtils.equalsIgnoreCase(normalise(addressDtoLine1), normalise(addressLine1))
```

Comment:
I wanted to match the style of the rest of the lines in that block.

That said, looking at it again honestly, it seems a bit difficult to read with that big block.

I could either create a method that encapsulates the `StringUtils.equalsIgnoreCase` and `normalise` like so

```
    private static boolean addressFieldsEqual(String field1, String field2){
        return StringUtils.equalsIgnoreCase(normalise(field1), normalise(field2));
    }
```

Or I could go further and make one for all of them like so 

```
    private static boolean addressFieldsEqual(String... args){
        if(args.length % 2 != 0){
            throw new IllegalArgumentException("Must have an even number of arguments");
        }
        for(int i = 0; i < args.length; i+=2){
            if(!StringUtils.equalsIgnoreCase(normalise(args[i]), normalise(args[i+1]))){
                return false;
            }
        }
        return true;
    }
```

This would make the actual return statement this which is much easier to read 

```
        return addressFieldsEqual(
                addressDtoLine1, addressLine1,
                addressDto.getLine2(), addressApi.getAddressLine2(),
                addressDto.getTown(), addressApi.getLocality(),
                addressDto.getCounty(), addressApi.getRegion(),
                addressDto.getCountry(), addressApi.getCountry(),
                addressDto.getPoBox(), addressApi.getPoBox(),
                addressDto.getCareOf(), addressApi.getCareOf(),
                addressDto.getPostcode(), addressApi.getPostalCode()
        );
```

Obviously, there is an issue with that Exception though that should be caught with Unit Tests
Otherwise, I could use `Pair`s though that seems like that may take away readability again

Reply:
Just saw that the same logic is used for the CompanyIdentification so might need a better name for that method.

`fuzzyStringPairsEqual`?

Reply:
My thought was that we should normalise the premises and line1 before combining them or it could result in 2 different strings when it shouldn't?

I'm not so worried about the (existing) duplication of code and wouldn't tackle it as part of this high priority bug

Reply:
The normalise method trims the strings so if we don't call it before combining them we would have mismatches.

DTO:
- property name number: "The house "
- line 1: " the street "

API:
- premises: "The house"
- line 1: "the street"

This would result in
- addressDtoLine1 = "The house   the street"
- addressLine1 = "The house the street"

Which aren't equals but I believe this should be treated as equals?

Reply:
Actually, it might work as normalise also reduces multiple whitespaces to just one